tiledb/sm/misc/types.h

```diff
@@ -318,6 +320,21 @@ class ByteVecValue {
     x_.clear();
   }
 
+  // To string function
+  std::string to_str() const {
```

Comment:
Maybe call this `to_hex_str()` to indicate it isn't printing a string of ascii characters.

Maybe also have some tests for `to_hex_str()` that exercise all higher-order bytes.